### PR TITLE
CMR-6284 Support searching by microversion by extension.

### PR DESCRIPTION
**Pull request by eereiter**

### CODE REVIEW

File: common-lib/src/cmr/common/mime_types.clj [hunk 1]
```diff
@@ -199,11 +199,13 @@
...
-   JSON mime type with the specified version."
+   umm_json_vN_N_N (_N can be repeated as many times as needed) where N is a version number separated
+   by underscores then it will return a UMM JSON mime type with the specified version."
   [extension]
-  (when-let [[_ major minor] (re-matches #"umm_json_v(\d+)_(\d+)" extension)]
-    (format "%s;version=%s.%s" umm-json major minor)))
+  (when (re-find #"^umm_json_v" extension)
+    (let [x (re-seq #"(\d+)" extension)
+          s (map #(% 1) x)]
+      (format "%s;version=%s" "application/vnd.nasa.cmr.umm+json" (clojure.string/join "." s)))))
```
Comment (eereiter) [inline common-lib/src/cmr/common/mime_types.clj:208]: I need to replace "application/vnd.nasa.cmr.umm+json" with umm-json

Comment (eereiter) [reply common-lib/src/cmr/common/mime_types.clj:208]: done.

**Comment by eereiter** [comment]
> Need to add tests verify both the success and failure scenarios when searching with micro versions.

Are the following not adequate? 
    (is (= (str mt/umm-json ";version=1")
           (mt/path->mime-type "granules.umm_json_v1")))
    (is (= (str mt/umm-json ";version=1.3.4.5.66")
           (mt/path->mime-type "granules.umm_json_v1_3_4_5_66")))
    (is (= nil
           (mt/path->mime-type "granules.umm_j1son_v1_3_4_5_66")))

I added these to the common-lib/test/cmr/common/test/mime_types.clj file in the pull request.

**Comment by ygliuvt** [comment]
I think we should add search by supported minor version (e.g. UMM-C v1.15.1) as a successful case with integration test and a search by unsupported minor version (e.g. UMM-C v1.14.3) as a failure case with integration test. The test you added in common lib are just unit test verify that you can convert the format suffix to mime type. There is no test covering search app is handling the minor version correctly. The search-umm-json-error-case test you referenced is a good place to just add a test case with UMM-C v1.14.3.

**Comment by eereiter** [comment]
@ygliuvt Thanks for explaining what tests are needed.  I have added some new tests.

**Pull request merged**